src/bp/core/services/action/local-action-server.ts

```diff
@@ -94,19 +86,30 @@ export class LocalActionServer {
     this.app.post(
       '/action/run',
       _validateRunRequest,
-      this.asyncMiddleware(async (req: RunActionRequest, res: RunActionResponse) => {
-        const { actionArgs, actionName, botId, token, incomingEvent } = req.body
-
-        const service = await this.actionService.forBot(botId)
-        try {
-          await service.runLocalAction({ actionName, actionArgs, incomingEvent, token, runType: 'http' })
-        } catch (e) {
-          throw new InternalServerError(e)
+      this.asyncMiddleware(
+        async (
+          req: TypedRequest<{
+            actionArgs: any
+            actionName: string
+            botId: string
+            token: string
+            incomingEvent: IO.IncomingEvent
+          }>,
```

Comment:
Can you move this in an interface, and then use it here and in the runActionLocal method? They are defined manually twice

Reply:
If you look closely, their parameters are not exactly the same.

Reply:
Then create a type which Pick or Omit the corresponding properties. This ensures consistency & avoid errors when refactoring

Reply:
Done in 55b9789